### ISSUE TITLE
XmlPullParserException in pdf2htmlEX-Android v0.18.20 

### ISSUE THREAD

**Issue by jackyhieu1211** [ViliusSutkus89/pdf2htmlEX-Android#83]
```
W/System.err: org.xmlpull.v1.XmlPullParserException: expected: END_TAG {null}family (position:END_TAG </axis tag='ital' stylevalue='0'>@28:45 in java.io.InputStreamReader@80377c5) 
W/System.err:     at com.android.org.kxml2.io.KXmlParser.require(KXmlParser.java:2067)
W/System.err:     at com.viliussutkus89.android.fontconfigtranslator.FontconfigTranslator$FontFamily.<init>(FontconfigTranslator.java:245)
W/System.err:     at com.viliussutkus89.android.fontconfigtranslator.FontconfigTranslator.translate(FontconfigTranslator.java:104)
W/System.err:     at com.viliussutkus89.android.pdf2htmlex.FontconfigAndroid.init(FontconfigAndroid.java:27)
W/System.err:     at com.viliussutkus89.android.pdf2htmlex.pdf2htmlEX.<init>(pdf2htmlEX.java:119)

```

It working on version v0.18.19. But I use version v0.18.20. It got error above.

This is my config 

           
     val outputFile =
                pdf2htmlEX(context)
                    .setInputPDF(inputPdf)
                    .setOwnerPassword(password)
                    .setUserPassword(password)
                    .setOutputHtml(outputPath)
                    .setEmbedCSS(true)
                    .setEmbedImage(true)
                    .setEmbedJavascript(true)
                    .setOptimizeText(true)
                    .setSqueezeWideGlyph(true)
                    .setStretchNarrowGlyph(true)
                    .setProcessAnnotation(true)
                    .setDRM(true)
                    .convert()
        

You can check this file

[pdf_file.pdf](https://github.com/ViliusSutkus89/pdf2htmlEX-Android/files/13785699/pdf_file.pdf)

**Comment by ViliusSutkus89** [comment]
Hi,

If it works on 0.18.19, it should work on 0.18.20, because the exception happens in the library that I haven't updated in between these two releases.

Anyways, lets see what we can do.
Could you download `/system/etc/fonts.xml` file from your device? It's a device config file that contains info about available fonts.

**Comment by ViliusSutkus89** [comment]
Thank you, @jackyhieu1211 . Could you also check if 0.18.21 works for you? I've added a try-catch for Fontconfig translator, there may be some issues with fonts, but the pdf conversion should work

**Comment by ViliusSutkus89** [comment]
Error is still there, because it's printed out in the line 31, but the conversion process should still continue, is it not doing anything?

**Comment by ViliusSutkus89** [comment]
It shouldn't have stopped. Could you post more output from logcat?

**Comment by ViliusSutkus89** [comment]
The error is not in libc, libc only raises segmentation fault, because there's an error in our code somewhere.

So this is a separate error from the XmlPullParserException, I've reproduced it on my end, will see how can I solve it, but the fix won't be today

**Comment by ViliusSutkus89** [comment]
OK, so your PDF has a font embedded with a name "BGDGBW弫Roboto-Bold".
FontForge isn't happy about non ASCII characters in font names and is trying to print out an error, but fails, because iconv is failing. I've patched FontForge to handle iconv failure ( https://github.com/ViliusSutkus89/ndkports/commit/fd0b11e1fde233ff16369a906ef44f9e5b947925 ).

Will later look into why iconv fails, because iconv may be used in other places too, without proper error handling.

That error message: "The PostScript font name \"BGDGBW弫Roboto-Bold\" is invalid.\nIt should be printable ASCII,\nmust not contain (){}[]<>%/ or space\nand must be shorter than 63 characters"

Also, please tell me that Google Translate is translating that symbol wrong and it doesn't mean "Kill Roboto Bold".

**Comment by ViliusSutkus89** [comment]
@ jackyhieu1211 , can you confirm is the issue is fixed for you in 0.18.22 ?

**Comment by ViliusSutkus89** [comment]
Yeah, conversion progress would be useful. Let me quote myself from more than three years ago in Issue #20:
>Totally reasonable expectation. Will get around to that someday

Haven't got around to it yet
